examples/chip-tool/commands/group/Commands.h

```diff
@@ -0,0 +1,384 @@
+
+    CHIP_ERROR RunCommand() override
+    {
+        fprintf(stderr, "\n");
+        fprintf(stderr, "  +-------------------------------------------------------------------------------------+\n");
+        fprintf(stderr, "  | Available Groups :                                                                  |\n");
+        fprintf(stderr, "  +-------------------------------------------------------------------------------------+\n");
+        fprintf(stderr, "  | Group Id   |  KeySet Id     |   Group Name                                          |\n");
+        chip::FabricIndex fabricIndex;
+        CurrentCommissioner().GetFabricIndex(&fabricIndex);
+        chip::Credentials::GroupDataProvider * groupDataProvider = chip::Credentials::GetGroupDataProvider();
+        auto it                                                  = groupDataProvider->IterateGroupInfo(fabricIndex);
+        chip::Credentials::GroupDataProvider::GroupInfo group;
+        if (it)
+        {
+            while (it->Next(group))
+            {
+                chip::KeysetId keysetId;
+                if (FindKeySetId(fabricIndex, group.group_id, keysetId))
+                {
+                    fprintf(stderr, "  | 0x%-12x  0x%-13x  %-50s |\n", group.group_id, keysetId, group.name);
+                }
+                else
+                {
+                    fprintf(stderr, "  | 0x%-12x  %-15s  %-50s |\n", group.group_id, "None", group.name);
+                }
+            }
+            it->Release();
+        }
+        fprintf(stderr, "  +-------------------------------------------------------------------------------------+\n");
+        SetCommandExitStatus(CHIP_NO_ERROR);
+        return CHIP_NO_ERROR;
+    }
+};
+
+class AddGroup : public CHIPCommand
+{
+public:
+    AddGroup(CredentialIssuerCommands * credsIssuerConfig) : CHIPCommand("add-group", credsIssuerConfig)
+    {
+        AddArgument("groupName", &groupName);
+        AddArgument("groupId", chip::kUndefinedGroupId, UINT16_MAX, &groupId);
+    }
+    chip::System::Clock::Timeout GetWaitDuration() const override { return chip::System::Clock::Seconds16(20); }
+
+    CHIP_ERROR RunCommand() override
+    {
+        CHIP_ERROR err = CHIP_NO_ERROR;
+        if (strlen(groupName) > CHIP_CONFIG_MAX_GROUP_NAME_LENGTH || groupId == chip::kUndefinedGroupId)
+        {
+            err = CHIP_ERROR_INVALID_ARGUMENT;
+        }
+        else
+        {
+            chip::FabricIndex fabricIndex;
+            CurrentCommissioner().GetFabricIndex(&fabricIndex);
+            chip::Credentials::GroupDataProvider * groupDataProvider = chip::Credentials::GetGroupDataProvider();
+            chip::Credentials::GroupDataProvider::GroupInfo group;
+
+            group.SetName(groupName);
+            group.group_id = groupId;
+            err            = groupDataProvider->SetGroupInfo(fabricIndex, group);
+        }
+        SetCommandExitStatus(err);
+        return err;
+    }
+
+private:
+    char * groupName;
```

Comment:
Should this be an optional arg?

Reply:
Well the GroupDataProvider API doesn't make it optional, and from an user point of view, the second you'll have more than 1 group, a name will be needed to avoid confusion.